source_code/main_mcu/src/NODEMGMT/nodemgmt.h

```diff
@@ -142,6 +142,19 @@ typedef struct
     uint8_t reserved2[6];           // Reserved for future use    
 } child_data_node_second_half_t;
 
+typedef struct TOTP_cred_node_s
+{
+    union
+    {
+        uint8_t TOTPsecret[TOTP_SECRET_MAX_LEN];    // Encrypted secret
+        uint8_t TOTPsecret_ct[TOTP_SECRET_MAX_LEN]; // Decrypted (cleartext) secret
+    };
+    uint16_t TOTPsecretLen;        // Length of TOTPsecret
```

Comment:
guessing we will know that this credential doesn't have any TOTP if the secret len is 0 or 0xFFFF

Reply:
Yes, when it is 0 since we zeroize the node.